core/drivers/crypto/caam/hal/imx_8q/hal_rng.c

```diff
@@ -18,3 +18,12 @@ enum caam_status caam_hal_rng_instantiated(vaddr_t baseaddr __unused)
 	else
 		return CAAM_NO_ERROR;
 }
+
+bool caam_hal_rng_pr_enabled(vaddr_t baseaddr __unused)
+{
+	/*
+	 * We have no direct access to the CAAM and can't assume prediction
+	 * resistance was enabled by the system controller firmware.
+	 */
+	return false;
```

Comment:
No need to reach for the firmware. You can read `RNG_STA_PRx` instead

Reply:
Can I use the implementation `caam_hal_rng_pr_enabled` implementation in `hal_rng.c` unchanged?

Reply:
Yes